### PR TITLE
OAuth 2.0 Troubleshooting guide updates per provided feedback

### PR DESCRIPTION
**Pull request by MarcialRosales**

### CODE REVIEW

**Review by michaelklishin — APPROVED** [review]

File: docs/troubleshooting-oauth2.md [hunk 1]
```diff
@@ -23,119 +23,116 @@ limitations under the License.
 
 ## Overview {#overview}
 
-This guide covers the most common errors encountered using [OAuth 2.0](./oauth2/) and management plugin and how to diagnose them.
+This guide covers the most common errors encountered using [OAuth 2.0](./oauth2/) and [management plugin](./management) and how to diagnose them.
```
Comment (pstack2021) [inline docs/troubleshooting-oauth2.md:27]: add the word "the" before management plugin

File: docs/troubleshooting-oauth2.md [hunk 2]
```diff
@@ -23,119 +23,116 @@ limitations under the License.
...
 
-#### Troubleshoot the issue
+#### Troubleshooting
 
 These are the most common reasons for this issue:
 
- * The endpoint is either down or is not reachable (e.g. there is a firewall which blocks access)
- * The endpoint has a TLS certificate not [trusted](./ssl#peer-verification) by the browser
- * The browser is blocking access due to [CORS](https://en.wikipedia.org/wiki/Cross-origin_resource_sharing) policy
+* The endpoint is unreachable (e.g. there is a firewall which blocks access) or target service is down
```
Comment (pstack2021) [inline docs/troubleshooting-oauth2.md:45]: use "for example" instead of abbreviations like e.g in docs

File: docs/troubleshooting-oauth2.md [hunk 3]
```diff
@@ -23,119 +23,116 @@ limitations under the License.
...
+#### Troubleshooting
 
+This issue is caused when the endpoint is not returning a JSON payload which matches with the [OpenId Connect Discovery Configuration](https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfig).
+These are the possible causes:
+- The payload returned by the endpoint is not compliant because it is empty or it is missing some critical information. To identify the root cause, open the browser's javascript console and search for one of these possible error messages:
+  - `Payload does not contain openid configuration` This error occurs when the payload is empty or it is not a JSON payload.
+  - `Missing authorization_endpoint` This error occurs when the JSON attribute `authorization_endpoint` is missing.
+  - `Missing token_endpoint` This error occurs when the JSON attribute `token_endpoint` is missing.
+  - `Missing jwks_uri` This error occurs when the JSON attribute `jwks_uri` is missing.
+- The url is wrong. Check out with your administrator of your Identity Provider to get the correct url to the OpenId Connect Discovery endpoint.
```
Comment (pstack2021) [inline docs/troubleshooting-oauth2.md:87]: remove the word "out" after "Check"

File: docs/troubleshooting-oauth2.md [hunk 4]
```diff
@@ -23,119 +23,116 @@ limitations under the License.
...
 
 ### Not authorized {#not-authorized-error}
 
-#### Steps to reproduce the issue
+#### Steps to reproduce
 
-Open the root url of the management UI in the browser, click on the buttoh "Click here to logon" and enter the credentials requested by your Identity Provider. You are redirected back to the management UI with the following error:
+Open the root url of the management UI in the browser, click on the button "Click here to logon" and enter the credentials requested by your Identity Provider. You are redirected back to the management UI with the following error:
```
Comment (pstack2021) [inline docs/troubleshooting-oauth2.md:94]: add full stop after the word "browser"
Then start a new sentence written as: "click the "Click here to logon" button and enter the credentials requested by your Identity Provider."

File: docs/troubleshooting-oauth2.md [hunk 5]
```diff
@@ -23,119 +23,116 @@ limitations under the License.
...
-1. Open your browwser's developer tool (e.g. in Chrome or Firefox, right-click on the page and click on *Inspect* menu option)
-2. Go to the tab *Application*
-3. Select the option *Storage* > *Local Storage* in the left panel
-4. Click on the tree option which matches the URL of the management UI
-5. Select the Key *rabbitmq.credentials* in the right panel
-6. Copy its value
-7. Go to the url https://jwt.io
-8. Paste the value into the text field *Encoded*
-9. Look at the payload's text field *Decoded*
+1. Open your browwser's developer tool (e.g. in Chrome or Firefox, right-click on the page and click on *Inspect* menu option).
```
Comment (pstack2021) [inline docs/troubleshooting-oauth2.md:109]: change e.g to "for example" and again I don't think there is need for brackets. I would start "For example..." as a new sentence

File: docs/troubleshooting-oauth2.md [hunk 6]
```diff
@@ -23,119 +23,116 @@ limitations under the License.
...
+2. Go to the tab *Application*.
```
Comment (pstack2021) [inline docs/troubleshooting-oauth2.md:110]: Go to the *Application* tab.

File: docs/troubleshooting-oauth2.md [hunk 7]
```diff
@@ -23,119 +23,116 @@ limitations under the License.
...
-#### Troubleshoot the issue
+#### Troubleshooting
 
-1. Access RabbitMQ [logs](./logging)
-2. Look up for `{bad_cert,hostname_check_failed}`
-3. Look for an entry it says that RabbitMQ tried to contact the url found in `auth_oauth2.issuer` and the issuer presented a wildcard certificate
-4. To fix this issue, add the following line to your RabbitMQ configuration `auth_oauth2.https.hostname_verification = wildcard`. Or `auth_oauth2.oauth_providers.<my_oauth_provider_name>.https.hostname_verification = wildcard` if you are using multiple OAuth providers
+1. Access RabbitMQ logs.
+2. Look for `{bad_cert,hostname_check_failed}`.
+3. If you find an entry it means that RabbitMQ tried to contact the url found in `auth_oauth2.issuer` and the issuer presented a wildcard certificate.
```
Comment (pstack2021) [inline docs/troubleshooting-oauth2.md:137]: url or URL? keep consistent through, check existing OSS RMQ docs to see what is used consistently and continue with that, thanks

Comment (pstack2021) [reply docs/troubleshooting-oauth2.md:137]: add a comma after "If you find and entry,...."

File: versioned_docs/version-3.13/troubleshooting-oauth2.md [hunk 8]
```diff
@@ -23,119 +23,116 @@ limitations under the License.
...
+4. You are redirected back to RabbitMQ but with an error "No Authorized".
 
-#### Troubleshoot the issue
+#### Troubleshooting
 
-1. Access RabbitMQ [logs](./logging)
-2. Look up for `{bad_cert,hostname_check_failed}`
-3. Look for an entry it says that RabbitMQ tried to contact the url found in `auth_oauth2.issuer` and the issuer presented a wildcard certificate
-4. To fix this issue, add the following line to your RabbitMQ configuration `auth_oauth2.https.hostname_verification = wildcard`. Or `auth_oauth2.oauth_providers.<my_oauth_provider_name>.https.hostname_verification = wildcard` if you are using multiple OAuth providers
+1. Access RabbitMQ logs.
```
Comment (pstack2021) [inline versioned_docs/version-3.13/troubleshooting-oauth2.md:135]: Add "the" before "RabbitMQ" above

File: versioned_docs/version-3.13/troubleshooting-oauth2.md [hunk 9]
```diff
@@ -23,119 +23,116 @@ limitations under the License.
...
+2. Look for `{bad_cert,hostname_check_failed}`.
+3. If you find an entry it means that RabbitMQ tried to contact the url found in `auth_oauth2.issuer` and the issuer presented a wildcard certificate.
```
Comment (pstack2021) [inline versioned_docs/version-3.13/troubleshooting-oauth2.md:137]: see previous comment

**Comment by michaelklishin** [comment]
@pstack2021 thank you, all addressed.

**Pull request merged**